src/main/java/com/fauna/client/FaunaClient.java

```diff
@@ -152,7 +153,7 @@ private <E> Supplier<CompletableFuture<FeedPage<E>>> makeAsyncFeedRequest(HttpCl
                         // Fall back on ProtocolException.
                         throw new ProtocolException(response.statusCode(), failure);
                     }
-                    return FeedPage.parseResponse(response, codec);
+                    return FeedPage.parseResponse(response, codec, statsCollector);
```

Comment:
This is only capturing stats on a 200 response. What about a non-200?

Reply:
In the E2E test case that I added, it looks like the stats do not get updated, but I refactored the error handling, so it follows the same path as query failures.